custom/enikshay/data_store.py

```diff
@@ -32,10 +32,11 @@ def latest_adherence_date(self, episode_id):
         if result:
             # the result is sorted on 'adherence_date'
             latest_date = result[0].get('adherence_date')
-            if not latest_date or not parse_date(latest_date):
+            parsed_date = parse(latest_date).date()
```

Comment:
Yeah, that's needed here . The empty argument issue is handled in the next line  `if not latest_date`. `parse_date` was not handling date strings with timezone info on them